### ISSUE TITLE
Unable to compile v0.10 - Errors in aliases_parse.y and dma.h

### ISSUE THREAD

**Issue by dallenwilson** [corecode/dma#35]
Disclaimer: I am working with an ancient sparc box, running a bastardized version of Debian Sarge. Many packages are several years (5+) old, and I acknowledge that I may be out of luck as far as support goes.

I'm trying to compile dma v0.10, and getting an error almost immediately. I believe it's YACC related, but I haven't found much via google to help me in narrowing it down. If not YACC, then it might just be that my copy of gcc is Too Damn Old.

I've installed the latest versions of both bison and flex (bison to v3.0.4 flex to v2.5.39). gcc is v3.3.5.

The output is below:
```bash
[warped@dark-core:~/Code/dma-0.10]$ make CC=/usr/bin/gcc YACC=/usr/local/bin/bison LEX=/usr/local/bin/flex SH=/bin/bash INSTALL=/usr/bin/install
/usr/bin/gcc -O -pipe -Wall -DDMA_VERSION='"v0.10"' -DLIBEXEC_PATH='"/usr/local/lib"' -DCONF_PATH='"/etc/dma"'  -include dfcompat.h -o aliases_parse.o -c aliases_parse.c
In file included from aliases_parse.y:6:
dma.h:96: error: syntax error before "SLIST_ENTRY"
dma.h:96: warning: no semicolon at end of struct or union
dma.h:98: error: syntax error before '}' token
dma.h:99: warning: type defaults to `int' in declaration of `SLIST_HEAD'
dma.h:99: warning: parameter names (without types) in function declaration
dma.h:99: warning: data definition has no type or storage class
dma.h:104: error: field `dests' has incomplete type
dma.h:147: error: syntax error before "SLIST_ENTRY"
dma.h:147: warning: no semicolon at end of struct or union
dma.h:151: error: syntax error before '}' token
dma.h:152: warning: type defaults to `int' in declaration of `SLIST_HEAD'
dma.h:152: warning: parameter names (without types) in function declaration
dma.h:152: warning: data definition has no type or storage class
aliases_parse.y: In function `yyparse':
aliases_parse.y:47: warning: implicit declaration of function `LIST_FIRST'
aliases_parse.y:47: error: invalid lvalue in assignment
aliases_parse.y:58: error: syntax error before "else"
aliases_parse.y:74: warning: implicit declaration of function `SLIST_FIRST'
aliases_parse.y:74: error: invalid lvalue in assignment
aliases_parse.y:89: error: dereferencing pointer to incomplete type
aliases_parse.y:92: error: dereferencing pointer to incomplete type
aliases_parse.y:101: error: dereferencing pointer to incomplete type
aliases_parse.y:104: error: dereferencing pointer to incomplete type
aliases_parse.y:105: warning: implicit declaration of function `SLIST_NEXT'
aliases_parse.y:105: error: `next' undeclared (first use in this function)
aliases_parse.y:105: error: (Each undeclared identifier is reported only once
aliases_parse.y:105: error: for each function it appears in.)
aliases_parse.y:105: error: invalid lvalue in assignment
make: *** [aliases_parse.o] Error 1
[warped@dark-core:~/Code/dma-0.10]$ 
```

Can anyone point me in the right direction?

**Comment by dallenwilson** [comment]
I should add that I have not investigated the reports of syntax errors (eg, `dma.h:147: warning: no semicolon at end of struct or union`) as it compiles just fine on my desktop machine, running Linux Mint 17.2. That's what led me to think it may be an issue with my gcc being out of date.

**Comment by corecode** [comment]
could you run gcc with -E and put the resulting file/output in a gist?

**Comment by dallenwilson** [comment]
I modified the Makefile to add -E to my CFLAGS; that seemed like the simplest way to do it. Here's the std/stderr output in a gist, as requested: https://gist.github.com/dallenwilson/7632c49e8abd77b41c74

I may have figured out the problem, though. My copy of /usr/include/sys/queue.h seems to be lacking any mention of SLIST_ENTRY. Once I get that corrected I bet dma will compile without any further issues.

**Comment by dallenwilson** [comment]
Yep, replacing my outdated sys/queue.h with the (slightly) newer one from my Mint box fixed the problem.

Now it seems that my installed copy of libresolv doesn't export symbols, preventing dynamic linking. Or so google leads me to believe. But that's a separate issue.

Thanks for your help!
